src/SignalR/common/Http.Connections/src/Internal/HttpConnectionManager.cs

```diff
@@ -137,32 +137,18 @@ private async Task ExecuteTimerLoop()
             Log.HeartBeatEnded(_logger);
         }
 
-        public async Task ScanAsync()
+        public Task ScanAsync()
         {
             // Scan the registered connections looking for ones that have timed out
             foreach (var c in _connections)
             {
-                HttpConnectionStatus status;
-                DateTimeOffset lastSeenUtc;
                 var connection = c.Value.Connection;
-
-                await connection.StateLock.WaitAsync();
-
-                try
-                {
-                    // Capture the connection state
-                    status = connection.Status;
-
-                    lastSeenUtc = connection.LastSeenUtc;
```

Comment:
This fixes a potential race. Since there were no memory barriers before, it was theoretically possible to observe an Inactive status with a LastSeenUtc set long before the connection last became inactive. There's also the possibility of tearing without a lock.